src/dump.c

```diff
@@ -555,35 +555,35 @@ static void jl_serialize_datatype(ios_t *s, jl_datatype_t *dt)
         return;
     }
 
-    size_t nf = jl_datatype_nfields(dt);
-    write_uint16(s, nf);
     write_int32(s, dt->size);
     int has_instance = !!(dt->instance != NULL);
-    write_uint8(s, dt->abstract | (dt->mutabl<<1) | (dt->pointerfree<<2) | (has_instance<<3) |
+    int has_layout = !!(dt->layout != NULL);
```

Comment:
it's a common idiom for turning a number into a 0 or 1 (http://stackoverflow.com/questions/10307281/c-operator-is-a-two-not)